sympy/functions/elementary/hyperbolic.py

```diff
@@ -209,6 +209,52 @@ class cosh(HyperbolicFunction):
     sinh, tanh, acosh
     """
 
+    @property
+    def is_positive(self):
+        from sympy import pi
+        arg = self.args[0]
+
+        if arg.is_real:
+            return True
+
+        re, im = arg.as_real_imag()
+        if im % pi == 0 or re == 0:
```

Comment:
I think your observations are correct.

I was hoping that we could return a symbolic expression in case of non-numbers. I still think that should be possible. What would happen if `x == 0` were replaced with `Eq(x, 0)`?

As to [12] above, the test should be `if Or(k < 1, k > 3) is True`, since almost anything nonempty will count as True. (And similarly with `(k < 1) | (k > 3)`.)